lib/chef/knife/supermarket_share.rb

```diff
@@ -151,11 +151,11 @@ def do_upload(cookbook_filename, cookbook_category, user_id, user_secret_filenam
 
       def tar_cmd
         unless @tar_cmd
-          @tar_cmd = "tar"
+          @tar_cmd = "tar --numeric-owner --uid 0 --gid 0"
```

Comment:
that'll break non-freebsd non-gnutar posix tar implementations (solaris definitely and probably aix)

Reply:
i'd suggest you just be explicit about case statements for linux, mac, bsd and other, but that is difficult because knife doesn't run ohai.  knife should probably at least run --minimal-ohai to get platform stuff working, but that might be a bit out of scope for you, but the lack of that makes this all hard to do right.

Reply:
also i think the 'E' flag for extended metadata should be used on old posix tar to address some of your comments.